changes/pr4623.yaml

```diff
@@ -0,0 +1,23 @@
+# An example changelog entry
+#
+# 1. Choose one (or more if a PR encompasses multiple changes) of the following headers:
+#   - feature
+#   - enhancement
+#   - task
+#   - fix
+#   - deprecation
+#   - breaking (for breaking changes)
+#
+# 2. Fill in one (or more) bullet points under the heading, describing the change.
+#    Markdown syntax may be used.
+#
+# 3. If you would like to be credited as helping with this release, add a
+#    contributor section with your name and github username.
+#
+# Here's an example of a PR that adds an enhancement
+
+task:
+  - "Add new prometheus task to push to gateway - [#4623](https://github.com/PrefectHQ/prefect/pull/4623)"
```

Comment:
The Prometheus docs seem to recommend against this pattern

> Usually, the only valid use case for the Pushgateway is for capturing the outcome of a service-level batch job. A "service-level" batch job is one which is not semantically related to a specific machine or job instance (for example, a batch job that deletes a number of users for an entire service).

But here we have a flow/task run which is a specific job instance. Thoughts? @mashun4ek

Reply:
I'm also a bit confused about how this is supposed to be used after looking at the examples in  https://github.com/prometheus/client_python#gauge

Can you show how this would be used in a flow to push metrics of interest?

Reply:
We are using that today to push metrics about the quality of our input data.

Imagine you have a workflow with a task that extracts the quality of the data, like distribution, expected values, missing values. We are pushing this result for each job to prometheus to monitor our data into the flow.

We are using that as well to send data about our ML model performance during training. So we are able to quantify how good it is.

This task are not mean to monitor prefect. 
let me add an example into the code.

Reply:
> The Prometheus docs seem to recommend against this pattern
> 
> > Usually, the only valid use case for the Pushgateway is for capturing the outcome of a service-level batch job. A "service-level" batch job is one which is not semantically related to a specific machine or job instance (for example, a batch job that deletes a number of users for an entire service).
> 
> But here we have a flow/task run which is a specific job instance. Thoughts? @mashun4ek

Yes, you're right about the `push` pattern which is against Prometheus core, but in some cases, we can't use the pull approach and have to use Pushgateway. In this case, short-lived flows can't expose metrics for Prometheus to scrape, so instead, those flows will push metrics for Pushgateway. Thank you for contribution @davzucky! And I agree, a small example would be great!

Reply:
I updated the code and added a simple code that push to pushgateway the percentage of null value for a dataframe. This is something we are already using to provide data quality metric to our validation workflow